pkg/splunk/enterprise/afwscheduler.go

```diff
@@ -1320,14 +1390,34 @@ func getSHCPlaybookContext(ctx context.Context, client splcommon.ControllerClien
 }
 
 // getLocalScopePlaybookContext returns the local scoped app install playbook context
-func getLocalScopePlaybookContext(ctx context.Context, installWorker *PipelineWorker, sem chan struct{}, podExecClient splutil.PodExecClientImpl) PlaybookImpl {
+func getLocalScopePlaybookContext(ctx context.Context, installWorker *PipelineWorker, sem chan struct{}, podExecClient splutil.PodExecClientImpl) *localScopePlaybookContext {
```

Comment:
why are we changing the return type here?

Reply:
I would say keep it consistent for all the different types of playbook contexts. Probably better to return PlaybookImpl for all of them so that in future it would be better to write UTs around these functions.